src/platform.ts

```diff
@@ -0,0 +1,81 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All Rights Reserved.
+ * See 'LICENSE' in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import * as os from 'os';
+import * as util from './util';
+import { LinuxDistribution } from './linuxDistribution';
+
+function platformToJreOs(platform: string) {
+    return {
+        'win32': 'windows',
+        'linux': 'linux',
+        'darwin': 'mac'
+    }[platform];
+}
+
+function archToJreArch(arch: string) {
+    return {
+        'x86_64': 'x64',
+        'x86': 'x32'
+    }[arch];
+}
```

Comment:
Nitpicking: slight risk of getting an undefined here. How about using an `enum`, and/or restricting what can be passed as an argument (using the enum keys, or typing them out explicitly)?